### PR TITLE
Dev -> Master for v2.3 release

### PR DESCRIPTION
**Pull request by drpatelh**

### CODE REVIEW

**Review by saramonzon — APPROVED** [review]

File: conf/modules_illumina.config
Comment (JoseEspinosa) [inline conf/modules_illumina.config:87]: I guess here is not a problem since the `MULTIQC` not aliased process does indeed set the same path, but maybe it would be better to alias it to `MULTIQC_MAIN` to avoid any issue?

File: conf/modules_nanopore.config
Comment (JoseEspinosa) [inline conf/modules_nanopore.config:250]: This will be a good example to see if it's being overwritten by `MULTIQC` config selector

File: modules.json
Comment (JoseEspinosa) [inline modules.json:4]: You will love this one :eyes: , there is a new version of all these modules after this PR has been merged https://github.com/nf-core/modules/pull/1261

**Review by JoseEspinosa — APPROVED** [review]
Just made some minor comments. Awesome work guys! :clap:

File: modules.json
Comment (drpatelh) [reply modules.json:4]: Bah, it can wait for the next release now that we have tested everything. Didn't even see that it was merged! Thought we were still discussing it 👀

File: conf/modules_nanopore.config
Comment (drpatelh) [reply conf/modules_nanopore.config:250]: Good point. I will check anyway but be great if you can double check the alias stuff that came up on #rnaseq won't be a problem here 🤔

Comment (drpatelh) [reply conf/modules_nanopore.config:250]: I think we are good. I ran this command in `subworkflows` and `workflows` and we don't have any aliases that are the same as the process name.

```
find ./ -type f -exec grep -H 'include' {} \; | grep " as "
```

Comment (JoseEspinosa) [reply conf/modules_nanopore.config:250]: I am running a small test to check if in the case of `MULTIQC,` which is the only case where this is happening, could raise any issue. Anyway, I don't think this could be a big issue.

**Comment by JoseEspinosa** [comment]
@drpatelh found this warnings when running `NXF_VER=21.10.3 nextflow run main.nf -profile test_nanopore,docker --gff false`:
 ```
WARN: There's no process matching config selector: .*:.*:.*:.*:TABIX_BGZIP
WARN: There's no process matching config selector: .*:.*:.*:.*:.*:TABIX_TABIX
WARN: There's no process matching config selector: .*:.*:.*:.*:.*:BCFTOOLS_STATS
 ```

File: conf/modules_nanopore.config
Comment (JoseEspinosa) [reply conf/modules_nanopore.config:250]: You are right! Dont need to change anything :smile:

File: docs/usage.md
Comment (heuermh) [inline docs/usage.md:385]: I'd suggest not including this as an option, as it allows for non-reproducibility

File: lib/WorkflowCommons.groovy
Comment (heuermh) [inline lib/WorkflowCommons.groovy:84]: This would make my algorithms teacher upset, better to use a `Set` in this case.  Will suggest this change elsewhere

File: docs/usage.md
Comment (drpatelh) [reply docs/usage.md:385]: We have had a number of users that just want to use the latest version and this allows them to do that 🤷🏽

**Review by heuermh — APPROVED** [review]
Nice work!

File: lib/WorkflowCommons.groovy
Comment (drpatelh) [reply lib/WorkflowCommons.groovy:84]: This will have to wait until the next release now! Minor change :)

File: docs/usage.md
Comment (heuermh) [reply docs/usage.md:385]: 🤦

File: lib/WorkflowCommons.groovy
Comment (heuermh) [reply lib/WorkflowCommons.groovy:84]: Yep, no problem.  There are more low hanging fruit than this for improving performance 😉

**Comment by drpatelh** [comment]
> @drpatelh found this warnings when running NXF_VER=21.10.3 nextflow run main.nf -profile test_nanopore,docker --gff false:

No way around this I'm afraid. It's because we initialise `params.gff` in `main.nf` and not `nextflow.config` so it isn't accessible to use in `modules.config` hence the warnings. We need to find a better solution to this problem. It's quite deeprooted.

**Comment by drpatelh** [comment]
Thanks guys!

**Pull request merged**
